src/System.ComponentModel.EventBasedAsync/tests/AsyncOperationTests.cs

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.ComponentModel;
+using System.Threading;
+using Xunit;
+
+namespace System.ComponentModel.EventBasedAsync
+{
+    public class AsyncOperationTests
+    {
+        private const int k_spinTimeoutSeconds = 20;
+
+        [Fact]
+        public static void Noop()
+        {
+            // Test that a simple AsyncOperation can be dispatched and completed via AsyncOperationManager
+            var operation = new TestAsyncOperation(op => {});
+            operation.Wait();
```

Comment:
This could cause problems for you.  If Xunit uses a single-threaded synchronization context, and these .Wait() calls are blocking that thread waiting for AsyncOperation's callback to be posted to and execute on this thread, this is going to deadlock.  A simple workaround would be to get off of the Xunit synchronization context while testing anything that has to do with SynchronizationContext.  You could do this in a few ways:
- You could use SynchronizationContext.Current and SynchronizationContext.SetSynchronization context to capture the current context, set the current context to null, and then set it back to the original context when you're done.
- Do the whole body of the test inside of a Task.Run, the result of which you then block on.  That way, the actual test happens on a thread pool thread that doesn't have the xunit synchronization context set.